src/server/frontend_wayland/wl_data_device.cpp

```diff
@@ -125,8 +153,46 @@ void mf::WlDataDevice::set_selection(std::optional<wl_resource*> const& source,
     }
 }
 
+void mf::WlDataDevice::start_drag(
+    std::optional<wl_resource*> const& source,
+    wl_resource* origin,
+    std::optional<wl_resource*> const& icon,
+    uint32_t serial)
+{
+    // "The [origin surface] and client must have an active implicit grab that matches the serial"
+    if (!origin)
```

Comment:
I don't *think* we can get here? The protocol doesn't specify this as `allow-none`, so I *think* this would be filtered out before here?

Reply:
Agreed, missed this when cleaning up the code